snitun/client/client_peer.py

```diff
@@ -119,14 +121,32 @@ async def start(
         )
 
         # Task a process for pings/cleanups
-        self._loop.create_task(self._handler())
+        assert not self._run_task, "SniTun connection already running"
+        self._run_task = self._loop.create_task(self._handler())
```

Comment:
This was an existing bug where the task could get garbage collected before it finished